File: Interviewbit/linked-lists/swap-list-nodes-in-pairs.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     public int val;
 *     public ListNode next;
 *     ListNode(int x) { val = x; next = null; }
 * }
 */
public class Solution {
    public ListNode swapPairs(ListNode A) {
        if(A.next==null) return A;
        ListNode p=A,q=A.next;
        int tmp;
        while(q!=null){
            tmp=p.val;
            p.val=q.val;
            q.val=tmp;
            p=p.next.next;
            if(q.next!=null)q=q.next.next;
            else q=null;
        }
        return A;
    }
}

File: Interviewbit/linked-lists/swap-list-nodes-in-pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
ListNode* Solution::swapPairs(ListNode* A) {
    
    if(A==NULL||A->next==NULL) return A;
    
    ListNode* p=A;
    ListNode* q=A->next;
    
    while(q){
        swap(p->val,q->val);
        if(q->next){
            p=q->next;
            q=q->next->next;
        }else q=q->next;
    }
    return A;
}
